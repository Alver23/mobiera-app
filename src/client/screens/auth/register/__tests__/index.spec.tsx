// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Text, Button } from 'react-native-elements';
import * as NavigationHooks from '@react-navigation/native';

// Under test file
import RegisterScreen from '../index';

// Mocks
jest.mock('@react-navigation/native');

describe('<RegisterScreen />', () => {
  let component: ShallowWrapper;
  const navigation: any = {
    goBack: jest.fn(),
  };

  beforeEach(() => {
    jest.spyOn(NavigationHooks, 'useNavigation').mockReturnValue(navigation);
    component = shallow(<RegisterScreen />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(component.find(Text).exists()).toBeTruthy();
  });

  it('should go to the back', () => {
    const onPress = component.find(Button).prop('onPress');
    onPress();

    expect(navigation.goBack).toHaveBeenCalledTimes(1);
  });
});
