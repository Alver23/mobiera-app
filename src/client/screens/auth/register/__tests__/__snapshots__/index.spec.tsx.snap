// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterScreen /> should save a snapshot of the component 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <Themed.Text
    h3={true}
    style={
      Object {
        "color": "#43484d",
        "paddingVertical": 8,
        "textAlign": "center",
      }
    }
  >
    Get started
  </Themed.Text>
  <RegisterForm />
  <Themed.Button
    onPress={[Function]}
    title="Log in"
    type="outline"
  />
</ScrollView>
`;
